src/components/flowersPage/hooks/useFlowerData.ts

```diff
@@ -0,0 +1,34 @@
+import { useQuery } from '@tanstack/react-query';
+import { useMemo } from 'react';
+import { FlowerSchemaType } from '../FlowersPage.types.ts';
+import { FlowerSchema } from '../FlowersPage.schema.ts';
+import { flowerQueryKey } from '../../../Backend.constants.ts';
+
+export const useFlowerData = () => {
+  const getData = async (): Promise<FlowerSchemaType['items'] | undefined> => {
```

Comment:
getData can be moved to upper scope to able export in future when needed. Also that function will be created everytime `useFlowerData` will be created, with funtion in upper scope only once.